File: src/main/java/com/github/timo_reymann/ansible_vault_integration/execution/vaultaction/AnsibleVaultEncryptAction.java
package com.github.timo_reymann.ansible_vault_integration.execution.vaultaction;

import com.github.timo_reymann.ansible_vault_integration.execution.AnsibleVaultWrapperCallFailedException;
import com.github.timo_reymann.ansible_vault_integration.util.AnsibleVaultedStringUtil;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiFile;

import java.util.Collections;
import java.util.List;

public class AnsibleVaultEncryptAction extends AnsibleVaultAction {
    private final String plain;

    public AnsibleVaultEncryptAction(Project project, PsiFile contextFile, String plain) {
        super(project, contextFile);
        this.plain = plain;
    }

    @Override
    protected String getActionName() {
        return "encrypt";
    }

    @Override
    protected String getStdin() {
        return plain;
    }

    @Override
    protected List<String> getParameters() {
        return Collections.singletonList("--output=-");
    }

    @Override
    public String execute() throws AnsibleVaultWrapperCallFailedException {
        return AnsibleVaultedStringUtil.addPrefix(super.execute());
    }
}

File: src/main/kotlin/de/timo_reymann/ansible_vault_integration/action/vaultaction/AnsibleVaultEncryptAction.kt
package de.timo_reymann.ansible_vault_integration.action.vaultaction

import de.timo_reymann.ansible_vault_integration.action.execution.AnsibleVaultWrapperCallFailedException
import de.timo_reymann.ansible_vault_integration.action.util.AnsibleVaultedStringUtil
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiFile

open class AnsibleVaultEncryptAction(
    project: Project,
    contextFile: PsiFile,
    override val stdin: String
) : AnsibleVaultAction(project, contextFile) {
    override val actionName: String
        get() = "encrypt"

    override val parameters: List<String>
        get() = listOf("--output=-")

    @Throws(AnsibleVaultWrapperCallFailedException::class)
    override fun execute(): String = AnsibleVaultedStringUtil.addPrefix(super.execute())
}
